src/libstore/sqlite.hh

```diff
@@ -15,13 +15,16 @@ struct SQLite
 {
     sqlite3 * db = 0;
     SQLite() { }
-    SQLite(const Path & path);
+    SQLite(const Path & path, bool create = true);
     SQLite(const SQLite & from) = delete;
     SQLite& operator = (const SQLite & from) = delete;
     SQLite& operator = (SQLite && from) { db = from.db; from.db = 0; return *this; }
     ~SQLite();
     operator sqlite3 * () { return db; }
 
+    /* Disable synchronous mode, set truncate journal mode. */
+    void isCache();
```

Comment:
Just came across this in the source code. `isCache` really doesn't speak well for what it is doing. Maybe just calling it `setTruncateJournalMode` would have been better?

Reply:
No, that's an implementation detail. "isCache" is a more declarative statement - i.e. "this database is a cache so do whatever is possible to make it fast at the expensive of reliability".

Reply:
At first glance I would agree that `setTruncateJournalMode` seems to expose an implementation details and is thus maybe not ideal (On the other hand you added a comment saying "Disable synchronous mode, set truncate journal mode" because it would be very hard to derive that otherwise i guess)

From my point of view `isCache` is equally bad: My expectation is that any function adhering to a `isSomething` naming scheme falls into the category of predicate functions: 
```
if (Something is satisfied) return true else return false
```
But `isCache` returns `void` and does some stuff that you cannot guess without reading the comment so i am not sure this is in fact very "_declarative_".

It's not entirely clear to me what the level of abstraction is here, and if `isCache` is part of some API or if it is only ever going to exist for `SQLite`. 

- If this will only ever exist in the context of SQLite then this *is* implementation specific already and then i'm not sure why this shouldn't actually say what it does.
- if this is something that different storage classes will implement then I think a name like `useAsCache` would be more appropriate.

WDYT?